src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/DiagnosticCounter.cs

```diff
@@ -67,38 +66,45 @@ public void Dispose()
         /// <summary>
         /// Adds a key-value metadata to the EventCounter that will be included as a part of the payload
         /// </summary>
-        internal void AddMetaData(string key, string value)
+        public void AddMetadata(string key, string value)
         {
-            lock (MyLock)
+            lock (m_Lock)
             {
-                _metaData.Add(key, value);
+                _metadata = _metadata ?? new Dictionary<string, string>();
+                _metadata.Add(key, value);
             }
         }
 
-        internal string DisplayName { get; set; }
+        public string DisplayName { get; set; }
 
-        #region private implementation
+        public string Name { get; }
+
+        public EventSource EventSource { get; }
 
-        internal readonly string _name;
+        #region private implementation
 
         private CounterGroup _group;
-        private Dictionary<string, string> _metaData;
-        internal EventSource _eventSource;
+        private Dictionary<string, string> _metadata;
 
         internal abstract void WritePayload(float intervalSec);
 
         // arbitrarily we use name as the lock object.  
-        internal object MyLock { get { return _name; } }
+        internal object m_Lock { get { return Name; } }
```

Comment:
Why is a property using the `m_` prefix? Is this intended?

Reply:
It's internal only so... I guess I can just use keep the name as `MyLock`.

Reply:
Can it be private? Are external classes taking a dependency on the property? Or through the class public interface?

Reply:
It's used in all of the derived classes of this class. For example in PollingCounter::
```
        internal override void WritePayload(float intervalSec)
        {
            lock (MyLock)
            {
                double value = 0;

```

Reply:
Then, should it be protected instead? So only derived classes can take this dependency?

Reply:
Yes. I can make it protected.